src/LiveComponent/src/EventListener/LiveComponentSubscriber.php

```diff
@@ -141,11 +142,19 @@ public function onKernelController(ControllerEvent $event): void
 
         $this->container->get(LiveComponentHydrator::class)->hydrate($component, $data);
 
-        // extra variables to be made available to the controller
-        // (for "actions" only)
-        parse_str($request->query->get('values'), $values);
-        $request->attributes->add($values);
         $request->attributes->set('_component', $component);
+
+        if (\is_string($queryString = $request->query->get('args'))) {
+            // extra variables to be made available to the controller
+            // (for "actions" only)
+            parse_str($queryString, $args);
+
+            foreach ($this->readLiveActionArgs($component, $action) as $parameter => $arg) {
```

Comment:
Instead of including this method in this class, let's move to an internal static method on `LiveArg`.

I'm not sure if this is the best practice but it's what I've done for the other attributes (ie [`AsLiveComponent`](https://github.com/symfony/ux/blob/2.x/src/LiveComponent/src/Attribute/AsLiveComponent.php#L38)). I've found it helps keep the code a bit neater and concise.